test/e2e/src/test-custom-types.js

```diff
@@ -1,20 +1,26 @@
 const test = require('ava')
 const { Address } = require('../../..')
 const { Ok, Err } = require('../../../lib/rust_types')
-const { clientFor } = require('./util')
+const { clientFor, clientForFromTest, clientFromConstructor } = require('./util')
 
 test.before(async t => {
   const { client, keypair, contractId } = await clientFor('customTypes')
   const publicKey = keypair.publicKey()
   const addr = Address.fromString(publicKey)
-  t.context = { client, publicKey, addr, contractId } // eslint-disable-line no-param-reassign
+  t.context = { client, publicKey, addr, contractId, keypair } // eslint-disable-line no-param-reassign
 });
 
 test('hello', async t => {
   const { result } = await t.context.client.hello({ hello: 'tests' })
   t.is(result, 'tests')
 })
 
+test('hello from constructor', async t => {
+  const { client } = await clientFromConstructor('customTypes')
```

Comment:
Given that this is the only test that uses `clientFromConstructor`, and might _always be_ the only such test, probably not worth extracting the logic to a function in `util`.